analyze/index.ts

```diff
@@ -131,13 +138,32 @@ export async function isValidEmail(
   }
 }
 
+type RequestDetectBotsFunction = (request: AnalyzeRequest) => Array<string>;
+
 export async function detectBot(
   context: AnalyzeContext,
   request: AnalyzeRequest,
   options: BotConfig,
+  detect?: RequestDetectBotsFunction,
 ): Promise<BotResult> {
   const { log } = context;
-  const coreImports = createCoreImports();
+  let convertedDetect;
+  if (typeof detect === "function") {
+    convertedDetect = (req: string) => {
+      let request: AnalyzeRequest;
+      try {
+        request = JSON.parse(req);
+      } catch {
+        throw new Error("object sent for detection was not a request");
+      }
+      return detect(request);
+    };
+  }
+
+  const coreImports = createCoreImports(
+    noOpDetectSensitiveInfo,
+    convertedDetect || noOpDetectBots,
+  );
```

Comment:
```suggestion
  let convertedDetect = noOpDetectBots;
  if (typeof detect === "function") {
    convertedDetect = (req: string) => {
      let request: AnalyzeRequest;
      try {
        request = JSON.parse(req);
      } catch {
        throw new Error("object sent for detection was not a request");
      }
      return detect(request);
    };
  }

  const coreImports = createCoreImports(
    noOpDetectSensitiveInfo,
    convertedDetect,
  );
```
Will this work with the types?

Reply:
Yes this works!